### ISSUE TITLE
Add action compressor

### ISSUE THREAD

**Issue by thinkh** [VisualStorytelling/provenance-core#7]
Actions might resolve each other (e.g., open/close, select/deselect) or could be combined (e.g., multiple subsequent single selections could be combined to one multi-selection).

I assume that the compressor is application specific, since it depends on the registered actions. However, we should provide a basic implementation and test cases in the core, since it influences the tree traversal.

**Comment by Tommos0** [comment]
Yes, good idea. So, given a sequence of actions, the compressor should return a new (shorter) sequence of actions that produce the same result, right?
